### DISCUSSION TITLE
RPM .spec file

### DISCUSSION THREAD

**Discussion by idiot-shinuki** [geoffreybennett/alsa-scarlett-gui#151] [Q&A] [answered]
Hi, I use this software on Fedora 41 and I would like to package it as a .rpm file for ease of updating/installation. I have the andaman toolchain (used by the Terra repository) on my system in order to automate maintaining a local repo of different apps I use which either aren't packaged as rpm or don't receive updates from upstream as frequently as I'd like.

My point of posting here is I'd like to put together a .spec file to pull the latest commit to this project from github, build the application against my distro, and package it as a .rpm for my local repo. If there has already been a .spec file written to use with rpmbuild, is it possible for me to base my local .spec file off of it?

I plan to do this for every application that I'd normally run `make/cmake install` for. I normally avoid flatpaks when possible because of how much of my storage it eats so I'd greatly appreciate any possible help with building a .rpm of this.

**Comment by geoffreybennett** [comment]
Hi @idiot-shinuki,

If you run `make rpm` from the top-level of the git checkout, that will get you most of the way there I think?

Regards,
Geoffrey.

**Reply by idiot-shinuki** [reply]
I'd have to look into that. The build systems I use requires a .spec file to outline the build sources, build dependencies, and runtime dependencies as well as the build process and install location. If it generates a .spec file, that honestly works and I should be able to drop it right into my anda projects. If it builds the entire rpm, I could probably get the info from the .rpm itself and make a .spec file based on that.

**Comment by geoffreybennett** [comment, marked as answer]
Check it out; there's a spec file template in the repo, and the Makefile creates a tar with the generated spec file that gets passed to `rpmbuild -tb`. It's pretty simple.

**Reply by idiot-shinuki** [reply]
Oh man, how did I miss that 😂

I guess I didn't expect it to be in the top level directory. I did notice that there's a github actions workflow for building a .deb and I kinda just looked through subdirectories of the repo trying to find it.

**Reply by idiot-shinuki** [reply]
One thing I will say is it doesn't seem to outline build dependencies and runtime dependencies in the spec file. I could probably look into the Makefile or something along the lines of that to figure out what those are. I'll make a .spec file that more closely follows guidelines and then get back to you with the result in case you plan on putting it to use in any way. 

By the way, thank you for all that you do for Focusrite users on linux such as myself. The first thing I do anytime I need to do a clean install of linux is delete the current alsa-ucm-conf and replace it with your fork that adds support for the Scarlett 4i4 3rd Gen that I use. Works like a charm.

**Reply by geoffreybennett** [reply]
Oh nice! Thank you, and you're welcome! :)

I think the spec file just needs:
```
BuildRequires: make gcc pkgconfig gtk4-devel alsa-lib-devel openssl-devel
Requires: scarlett2-firmware
```

**Reply by idiot-shinuki** [reply]
That definitely helps a lot! Thank you for putting that together. I won't have time to start putting anything together until maybe tomorrow or the day after because I have work and I'm doing this as a side hobby but that saves me a lot of trouble, I really appreciate it!

**Reply by idiot-shinuki** [reply]
I forgot to include a summary in the spec file and didn't know why `dnf builddep alsa-scarlett-gui.spec` was throwing the error `Segmentation fault` with no additional info, but luckily I tried dnf4 (instead of the default dnf 5) and it told me what I was doing wrong.

Finally got around to listing them in a way that's searchable across different naming conventions for different distros and ended up with this:
```
BuildRequires:  make
BuildRequires:  gcc
BuildRequires:  pkgconfig
BuildRequires:  pkgconfig(gtk4)
BuildRequires:  pkgconfig(alsa)
BuildRequires:  pkgconfig(openssl)
```

I ran a quick search using dnf for scarlett2-firmware though and nothing came up. Could I ask where exactly that came from? If it needs to be present on the system for the full functionality of alsa-scarlett-gui then I could look into packaging that as well. As long as both are in my local repo then marking it as a required package should just work™

**Reply by geoffreybennett** [reply]
https://github.com/geoffreybennett/scarlett2-firmware

**Reply by idiot-shinuki** [reply]
I'm back, and I have everything in a state that I'm pleased with.

[alsa-scarlett-gui.tar.gz](https://github.com/user-attachments/files/17951427/alsa-scarlett-gui.tar.gz)
[scarlett2-firmware.tar.gz](https://github.com/user-attachments/files/17951428/scarlett2-firmware.tar.gz)

These files can be submitted to https://github.com/terrapkg/packages/ in order for people with the Terra repo enabled or people who use the Ultramarine distro to be able to install alsa-scarlett-gui and scarlett2-firmware using `sudo dnf install alsa-scarlett-gui`

I know nothing about the process for submitting packages to be built, I just use their build system for building/packaging things locally. I can go ahead and start that on my next day off from work if you'd like for me to look into that process and submit these two.
